pkg/storage/storage.go

```diff
@@ -0,0 +1,73 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type Profile struct {
+	Name string `json:"name"`
+	Key  string `json:"key"`
+}
+
+type Storage struct {
```

Comment:
Whether to keen this structure exported or not depends on the abstraction structure you choose.

First approach is about to keep our interfaces on the consumer side so you have to return exported structures from your constructors

Second approach is about creating interface in dedicated but yet common package and return the interface from the constructor. In that case your structure is preferred to be unexported to rely on interface but partictular structure.

And on more thing: do you actually expect this structure fields to be exported (if you will add constructor of course)?